onadata/libs/serializers/project_serializer.py

```diff
@@ -228,15 +227,21 @@ def get_users(self, obj):
                          self.context,
                          owner_query_param_in_request)
 
-    @profile("get_project_forms.prof")
     @check_obj
     def get_forms(self, obj):
+        forms = cache.get('{}{}'.format(PROJ_FORMS_CACHE, obj.pk))
```

Comment:
Yes, whenever permissions change or a form is deleted. Check [`XForm`]( https://github.com/onaio/onadata/blob/master/onadata/apps/logger/models/xform.py#L945) model